### PR TITLE
Various additional tests and benchmarks

### PR DESCRIPTION
**Pull request by theopolis**
1. Check that binaries are build without AVX extensions.
2. Require all pack/platform queries to execute non-1.
3. Add basic SQL and Event-based benchmarks.

### CODE REVIEW

File: tools/build.sh [hunk 1]
```diff
@@ -9,6 +9,7 @@
 
 set -e
 
+BUILD_RELEASE=True
```
Comment (mofarrell) [inline tools/build.sh]: Does this make the build a release build?  That will break the kernel test.

Comment (theopolis) [inline tools/build.sh]: Maybe that's a bad variable name. It was just ment to trigger additional tests. I'll take a look!

**Comment by theopolis** [comment]
Ok, there is an initial failure with OS X, this is good! See https://jenkins.osquery.io/job/osqueryPullRequestBuild/1684/TargetSystem=osx10/console for an example of a failed build due to AVX instructions present in the output binary.

**Comment by theopolis** [comment]
ok to test

**Comment by theopolis** [comment]
Now we begin an onslaught of tests and modifications to the build hosts. When OS X builds without AVX we snapshot the hosts as the restore target for `make deps`.

**Comment by theopolis** [comment]
A reinstall of libsnappy reduces the `vxorps` calls to 1: https://jenkins.osquery.io/job/osqueryPullRequestBuild/1685/TargetSystem=osx10/console

**Comment by theopolis** [comment]
Now down with `lz4`!

**Pull request merged**